pkg/handlers/metadata.go

```diff
@@ -104,7 +104,20 @@ func GetMetadataHandler(getK8sInfoFn MetadataK8sFn, kotsStore store.Store) http.
 			return
 		}
 		application := obj.(*kotsv1beta1.Application)
-		metadataResponse.IconURI = application.Spec.Icon
+
+		appIconURI, err := kotsStore.GetLatestAppIconURI(appID)
```

Comment:
if i'm understanding things correctly, this gets the app icon for the currently deployed version, not the latest one. also, i think the better approach to just updated the branding configmap when a version gets deployed, then there's no changes needed to this handler.

Reply:
>the branding configmap
there is no branding configmap.
we have `kotsadm-application-metadata` CM which holds `application.yaml` which always points to the latest downloaded version CM[which we cannot use]